apps/glusterfs/cluster_entry.go

```diff
@@ -168,5 +168,51 @@ func (c *ClusterEntry) NodeDelete(id string) {
 }
 
 func ClusterEntryUpgrade(tx *bolt.Tx) error {
+	err := addBlockFileFlagsInClusterEntry(tx)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func addBlockFileFlagsInClusterEntry(tx *bolt.Tx) error {
+
+	entry, err := NewDbAttributeEntryFromKey(tx, DB_HAS_FILE_BLOCK_FLAG)
+	// This key won't exist if we are introducing the feature now
+	if err != nil && err != ErrNotFound {
+		return err
+	}
+
+	if err == ErrNotFound {
+		entry = NewDbAttributeEntry()
+		entry.Key = DB_HAS_FILE_BLOCK_FLAG
+		entry.Value = "no"
+	} else {
+		//This case is only for future, if ever we want to set this key to "no"
```

Comment:
space between `//` and `This`

Reply:
One point is that if `Value == "yes"`, this means that the block/file flags have been present at previous starts of the software on this DB, and hence have been initialized. ==> we can skip this upgrade step.

Reply:
fixed first comment.

We do skip the upgrade step if value is yes. This check exists for future deprecation.